pkg/config/derive_test.go

```diff
@@ -0,0 +1,219 @@
+		t.Run(tt.want, func(t *testing.T) {
+			if got := d.MasterLBCNamePrefix(&tt.cs); got != tt.want {
+				t.Errorf("derived.MasterLBCNamePrefix() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDerivedRouterLBCNamePrefix(t *testing.T) {
+	tests := []struct {
+		cs   api.OpenShiftManagedCluster
+		want string
+	}{
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{
+					RouterProfiles: []api.RouterProfile{
+						{
+							FQDN: "one.two.three",
+						},
+						{
+							FQDN: "not.this",
+						},
+					},
+				},
+			},
+			want: "one",
+		},
+	}
+	d := derived{}
+	for _, tt := range tests {
+		t.Run(tt.want, func(t *testing.T) {
+			if got := d.RouterLBCNamePrefix(&tt.cs); got != tt.want {
+				t.Errorf("derived.RouterLBCNamePrefix() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDerivedPublicHostname(t *testing.T) {
+	tests := []struct {
+		cs   api.OpenShiftManagedCluster
+		want string
+	}{
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{FQDN: "bar", PublicHostname: "baar"},
+			},
+			want: "baar",
+		},
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{FQDN: "bar", PublicHostname: ""},
+			},
+			want: "bar",
+		},
+	}
+	d := derived{}
+	for _, tt := range tests {
+		t.Run(tt.want, func(t *testing.T) {
+			if got := d.PublicHostname(&tt.cs); got != tt.want {
+				t.Errorf("derived.PublicHostname() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDerivedKubeAndSystemReserved(t *testing.T) {
+	tests := []struct {
+		cs   api.OpenShiftManagedCluster
+		role acsapi.AgentPoolProfileRole
+		want string
+	}{
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{
+					AgentPoolProfiles: []api.AgentPoolProfile{
+						{
+							Name:   "1",
+							Role:   acsapi.AgentPoolProfileRoleCompute,
+							VMSize: "Standard_D2s_v3",
+						},
+					},
+				},
+			},
+			want: "cpu=200m,memory=512Mi",
+			role: acsapi.AgentPoolProfileRoleCompute,
+		},
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{
+					AgentPoolProfiles: []api.AgentPoolProfile{
+						{
+							Name:   "1",
+							Role:   acsapi.AgentPoolProfileRoleCompute,
+							VMSize: "Standard_D2s_v3",
+						},
+						{
+							Name:   "2",
+							Role:   acsapi.AgentPoolProfileRoleInfra,
+							VMSize: "Standard_D2s_v3",
+						},
+					},
+				}},
+			want: "cpu=200m,memory=512Mi",
+			role: acsapi.AgentPoolProfileRoleInfra,
+		},
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{
+					AgentPoolProfiles: []api.AgentPoolProfile{
+						{
+							Name:   "infra",
+							Role:   acsapi.AgentPoolProfileRoleMaster,
+							VMSize: "unknown",
+						},
+					},
+				},
+			},
+			want: "",
+			role: acsapi.AgentPoolProfileRoleMaster,
+		},
+		{
+			cs: api.OpenShiftManagedCluster{
+				Properties: &api.Properties{
+					AgentPoolProfiles: []api.AgentPoolProfile{},
+				},
+			},
+			want: "",
```

Comment:
it would be better to change the signature of {System,Kube}Reserved to return (string, error) and return an appropriate error if we're going to be testing this.